pkg/virt-controller/watch/vm_test.go

```diff
@@ -208,15 +208,32 @@ var _ = Describe("VirtualMachine", func() {
 			close(stop)
 		})
 
-		shouldExpectGracePeriodPatched := func(expectedGracePeriod int64, vmi *v1.VirtualMachineInstance) {
+		// TODO: We need to make sure the action was triggered
+		shouldExpectGracePeriodPatched := func(expectedGracePeriod int64) {
 			patch := fmt.Sprintf(`{"spec":{"terminationGracePeriodSeconds": %d }}`, expectedGracePeriod)
-			vmiInterface.EXPECT().Patch(context.Background(), vmi.Name, types.MergePatchType, []byte(patch), metav1.PatchOptions{}).Return(vmi, nil)
+			virtFakeClient.PrependReactor("update", "virtualmachineinstance", func(action testing.Action) (handled bool, ret runtime.Object, err error) {
+				switch action := action.(type) {
+				case testing.PatchActionImpl:
+					Expect(action.Patch).To(Equal([]byte(patch)))
+				default:
+					Fail("Expected to see patch")
+				}
+				return false, nil, nil
+			})
 		}
-
-		shouldExpectVMIFinalizerRemoval := func(vmi *v1.VirtualMachineInstance) {
+		// TODO: We need to make sure the action was triggered
```

Comment:
Is it not enough to check the vmi resource after the controller execution?

Reply:
I don't think so because the finalizer is not respected by the fake client.  Something I prefer to take as follow up.